File: geotrellis-python/tests/geotrellis_test/spark/io/file/FileSpatialSpec.py
from __future__ import absolute_import
from spec import Spec
from nose import tools
from tests.geotrellis_test.spark.io.PersistenceSpec import _PersistenceSpec
from tests.geotrellis_test.spark.io.SpatialKeyIndexMethods import _SpatialKeyIndexMethods
from tests.geotrellis_test.spark.TestEnvironment import _TestEnvironment
from tests.geotrellis_test.spark.testfiles.TestFiles import _TestFiles
from geotrellis.spark.SpatialKey import SpatialKey
from geotrellis.raster.Tile import Tile
from geotrellis.raster.FloatArrayTile import FloatArrayTile
from geotrellis.spark.TileLayerMetadata import TileLayerMetadata
from geotrellis.spark.io.file.FileLayerReader import FileLayerReader
from geotrellis.spark.io.file.FileLayerWriter import FileLayerWriter
from geotrellis.spark.LayerId import LayerId
from geotrellis.spark.io.index.ZCurveKeyIndexMethod import ZCurveKeyIndexMethod
from geotrellis.spark.io.file.FileValueReader import file_value_reader

@tools.nottest
class FileSpatialSpec(
        Spec,
        # TODO Tile doesn't have implicits to get AvroRecordCodec from
        #_PersistenceSpec[SpatialKey, Tile, TileLayerMetadata[SpatialKey]],
        _PersistenceSpec[SpatialKey, FloatArrayTile, TileLayerMetadata[SpatialKey]],
        _SpatialKeyIndexMethods,
        _TestFiles,
        _TestEnvironment,
        object):

    @property
    def reader(self):
        return FileLayerReader(self.sc, self.outputLocalPath)

    @property
    def writer(self):
        return FileLayerWriter(self.outputLocalPath)

    @property
    def tiles(self):
        return file_value_reader(self.outputLocalPath)

    @property
    def sample(self):
        return self.AllOnesTestFile

    class inner(object):
        "Filesystem layer names"

        def test_bad_characters(self):
            "should not throw with bad characters in name"
            layer = self.AllOnesTestFile
            layerId = LayerId("Some!layer:%@~`{}id", 10)
            print(self.outputLocalPath)
            #self.writer.write(SpatialKey, Tile, TileLayerMetadata, layerId, layer, ZCurveKeyIndexMethod.spatialKeyIndexMethod())
            self.writer.write(SpatialKey, FloatArrayTile, TileLayerMetadata[SpatialKey], layerId, layer, ZCurveKeyIndexMethod.spatialKeyIndexMethod())
            #backin = self.reader.read(SpatialKey, Tile, TileLayerMetadata, layerId)
            backin = self.reader.read(SpatialKey, FloatArrayTile, TileLayerMetadata[SpatialKey], layerId)

File: spark/src/test/scala/geotrellis/spark/store/file/FileSpatialSpec.scala
/*
 * Copyright 2016 Azavea
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package geotrellis.spark.store.file

import geotrellis.layer.*
import geotrellis.raster.Tile
import geotrellis.store.*
import geotrellis.store.file.*
import geotrellis.store.index.*
import geotrellis.spark.store.*
import geotrellis.spark.testkit.*
import geotrellis.spark.testkit.io.*
import geotrellis.spark.testkit.testfiles.TestFiles
import geotrellis.util.identityComponent
import org.apache.spark.rdd.RDD

class FileSpatialSpec
    extends PersistenceSpec[SpatialKey, Tile, TileLayerMetadata[SpatialKey]]
    with SpatialKeyIndexMethods
    with TestEnvironment
    with TestFiles
    with AllOnesTestTileSpec {
  lazy val reader: FileLayerReader = FileLayerReader(outputLocalPath)
  lazy val creader: FileCollectionLayerReader = FileCollectionLayerReader(outputLocalPath)
  lazy val writer: FileLayerWriter = FileLayerWriter(outputLocalPath)
  lazy val deleter: FileLayerDeleter = FileLayerDeleter(outputLocalPath)
  lazy val copier = FileLayerCopier(outputLocalPath)
  lazy val mover  = FileLayerMover(outputLocalPath)
  lazy val reindexer = FileLayerReindexer(outputLocalPath)
  lazy val tiles: FileValueReader = FileValueReader(outputLocalPath)
  lazy val sample: RDD[(SpatialKey, Tile)] with Metadata[TileLayerMetadata[SpatialKey]] = AllOnesTestFile

  describe("Filesystem layer names") {
    it("should not throw with bad characters in name") {
      val layer = AllOnesTestFile
      val layerId = LayerId("Some!layer:%@~`{}id", 10)

      println(outputLocalPath)
      writer.write[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](layerId, layer, ZCurveKeyIndexMethod)
      val backin = reader.read[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](layerId)
    }
  }
}
